scico/linop/_diag.py

```diff
@@ -156,10 +156,137 @@ def norm(self, ord=None):  # pylint: disable=W0622
             mord = snp.inf
         if mord not in ordfunc:
             raise ValueError(f"Invalid value {ord} for parameter ord.")
-        return ordfunc[mord](self.diagonal)
+        return ordfunc[mord](self._diagonal)
+
+
+class ScaledIdentity(Diagonal):
+    """Scaled identity operator."""
+
+    def __init__(
+        self,
+        scalar: float,
+        input_shape: Union[Shape, BlockShape],
+        input_dtype: DType = snp.float32,
+        **kwargs,
+    ):
+        """
+        Args:
+            scalar: Scaling of the identity.
+            input_shape: Shape of input array.
+            input_dtype: `dtype` of input argument.
+        """
+        if is_nested(input_shape):
+            diagonal = scalar * snp.ones(((),) * len(input_shape), dtype=input_dtype)
+        else:
+            diagonal = scalar * snp.ones((), dtype=input_dtype)
+        super().__init__(
+            diagonal=diagonal,
+            input_shape=input_shape,
+            input_dtype=input_dtype,
+            **kwargs,
+        )
+
+    def _eval(self, x: Union[Array, BlockArray]) -> Union[Array, BlockArray]:
+        return self._diagonal * x
```

Comment:
Not obvious why we need this instead of the `_eval` in `Diagonal`. Maybe a comment to clarify?

Reply:
Inclusion was an oversight. Removed.